frontend/projects/ui/src/app/services/dep-error.service.ts

```diff
@@ -0,0 +1,210 @@
+import { Injectable } from '@angular/core'
+import { Emver } from '@start9labs/shared'
+import { map, shareReplay } from 'rxjs/operators'
+import { PatchDB } from 'patch-db-client'
+import {
+  DataModel,
+  HealthCheckResult,
+  HealthResult,
+  InstalledPackageDataEntry,
+  PackageMainStatus,
+} from './patch-db/data-model'
+
+export type PackageDependencyErrors = Record<string, DependencyErrors>
+export type DependencyErrors = Record<string, DependencyError | null>
+
+@Injectable({
+  providedIn: 'root',
+})
+export class DepErrorService {
```

Comment:
🤔 So, other possible solutions that I can think of. 
For the finding the depErrors, you could have done the truly recursive, and that is repulsive. Sounds like you already explored it. 
Your [dynamic solution ](https://www.geeksforgeeks.org/dynamic-programming/) seems to work.

The downside is that we are doing a recompute every change. We could do a memoization on if the pkgs are the same object before, but that requires that we have a quick equal. Or the other is that the pointer equal would work, but then we need a guarentee that the pkgs are updated via a immutable style. I wouldn't trust this beyond using something like immutablejs 
Ok, the other is do the memoization deeper into the values that are easier run the equality checks. So we would could do a way of doing per pkg id, a combined observable based on each variable that we could do... Sounds like the the rxjs would make this a pain.
There are observable that might work, like [mobx](https://mobx.js.org/observable-state.html). It solves the diamond problem better, but the library would be new and unlike the rest of the thing. But it would do the leg work of the nested composted observable into something that feels natural. 

In summary, solution seems good. But the only downside is that we are not efficient at all (at least it looks like it)